theano/tensor/nnet/corr.py

```diff
@@ -706,36 +721,36 @@ def make_node(self, img, topgrad, shape=None):
 
     def infer_shape(self, node, input_shape):
         if self.border_mode == "half":
-            padH = padW = -1
+            padH_l = padH_r = padW_l = padW_r = -1
         elif self.border_mode == "full":
-            padH = padW = -2
+            padH_l = padH_r = padW_l = padW_r = -2
         elif isinstance(self.border_mode, tuple):
-            padH, padW = self.border_mode
+            (padH_l, padH_r), (padW_l, padW_r) = self.border_mode
```

Comment:
This breaks backward compatibility, since `self.border_mode` for older, pickled ops might still be `(int, int)`.

Reply:
And why isn't this using `get_conv_gradweights_shape` anyway?

Reply:
I thought this is fine since we normalize the border_mode in the BaseCorrMM.__init__ function. But I'll make the change.

Reply:
That's true, `BaseCorrMM__init__` covers most of the cases, but not the case where a graph that was pickled with the old version is unpickled in the new version. Unpickling doesn't run __init__ again, it directly restores the internal state of the object.